### PR TITLE
Add minimumPrice to BUYProduct

### PR DESCRIPTION
**Pull request by bgulanowski**
This adds a new local-only property to BUYProduct. This enables easier sorting of products in collections. This improves parity with the Android SDK, which already has this property.

@davidmuzi @gabrieloc @dbart01

### CODE REVIEW

File: Mobile Buy SDK/Mobile Buy SDK/Models/Persistent/BUYProduct.m [hunk 1]
```diff
@@ -35,6 +35,19 @@ @implementation BUYProduct
 
 @synthesize stringDescription=_stringDescription;
 
+- (void)setVariants:(NSOrderedSet *)variants
+{
+	[self willChangeValueForKey:BUYProductRelationships.variants];
+	NSMutableOrderedSet *primitiveVariants = [self primitiveVariants];
```
Comment (davidmuzi) [inline Mobile Buy SDK/Mobile Buy SDK/Models/Persistent/BUYProduct.m]: crashes in Advanced Sample App here.  Unrecognized selector

**Comment by davidmuzi** [comment]
Should also have a unit test

File: Mobile Buy SDK/Mobile Buy SDK/Models/Persistent/BUYProduct.m [hunk 2]
```diff
@@ -73,6 +86,20 @@ - (NSString *)stringDescription
...
+	NSDecimalNumber *newMinimumPrice = [self.variants valueForKeyPath:@"@min.price"];
+	if (![self.minimumPrice isEqual:newMinimumPrice]) {
+		self.minimumPrice = newMinimumPrice;
+	}
+}
+
+- (void)willSave
+{
+	[super willSave];
+	[self updateMinimumPrice];
```
Comment (dbart01) [inline Mobile Buy SDK/Mobile Buy SDK/Models/Persistent/BUYProduct.m]: `updateMinimumPrice` is called from `setVariants:` as well. Curious to know if there's ever a case where the price won't be updated if we eliminate `willSave`.

**Comment by bgulanowski** [comment]
A test is failing because this inadvertently assumes the Core Data environment, which tests do not.

Closing while I address this.

**Comment by bgulanowski** [comment]
In answer to @dbart01 's question, `-setVariants:` is not called when updating the object as part of deserialization. Not sure why. May only affect objects that inherit from NSObject. Will investigate that further, since that is definitely what we want, and to avoid relying on `-willSave` and `-didSave`.

**Pull request closed without merging**